aip/general/0193.md

```diff
@@ -62,11 +62,120 @@ a [`PreconditionFailure`][PreconditionFailure].
 **Note:** `ErrorInfo` represents a special case. There **must** be exactly one
 `ErrorInfo`. It is required.
 
-### Localization
+### Error messages
 
-Error messages **must** be in English. If a localized error message is also
-required, the service **should** use [`google.rpc.LocalizedMessage`][details]
-in the `details` field.
+For each error, the service **must** populate the `message` field on
+[`google.rpc.Status`][Status]. This error message,
+
+- is a developer-facing, human-readable "debug message"
+- is presented in the service's native language
+- both explains the error and offers an actionable resolution to it
+  ([citation](https://cloud.google.com/apis/design/errors#error_model))
+
+**Note:** Sometimes a service will elect to always present
+`Status.message` in English rather than the application's native
+language so that messages are easily searchable in common knowledge
+bases, such as StackOverflow&trade;.
+
+When introducing a new error, a failure scenario that did not previously
+occur for the service, the payload **must** include `ErrorInfo` and any
+variables found in dynamic segments of the error message **must** be
+present in `ErrorInfo.metadata`. See, "[Dynamic
+variables](#dynamic-variables)".
+
+#### Changing error messages
+
+If your service's API design governing body decides that altering
+`Status.message` represents a backwards incompatible change, then
+`Status.message` **must not** change.
+
+<a name="status-message-warning"></a>
+**Warning:** **Use extreme caution** when changing `Status.message`.
+API consumers often treat this error message as **part of the API
+contract**. Clients perform string matches on the text to differentiate
+one error for another and even parse the error message to extract
+variables from dynamic segments.
+
+There is a safer alternative. The service can chose to include an error
+message by adding [`google.rpc.LocalizedMessage`][LocalizedMessage] to
+[`Status.details`][Status details].
+
+The error message presented in `LocalizedMessage.message` **may** be the
```

Comment:
is this sentence stating something meaningful? if it *may* do something and *may* not do it, then that's all the options available and this is basically saying either is acceptable. Which feels similar to not including specific guidance at all.

Reply:
if it is relevant: is it possible to clarify why one would choose one over the other?

Reply:
See below. It is explained.